src/equations/compressible_euler_2d.jl

```diff
@@ -885,29 +885,49 @@ end
 
 # Calculate maximum wave speed for local Lax-Friedrichs-type dissipation as the
 # maximum velocity magnitude plus the maximum speed of sound
-# TODO: This doesn't really use the `orientation` - should it?
 @inline function max_abs_speed_naive(u_ll, u_rr, orientation::Integer, equations::CompressibleEulerEquations2D)
-  rho_ll, rho_v1_ll, rho_v2_ll, rho_e_ll = u_ll
-  rho_rr, rho_v1_rr, rho_v2_rr, rho_e_rr = u_rr
+  rho_ll, rho_v1_ll, rho_v2_ll, _ = u_ll
+  rho_rr, rho_v1_rr, rho_v2_rr, _ = u_rr
 
-  # Calculate primitive variables and speed of sound
-  v1_ll = rho_v1_ll / rho_ll
-  v2_ll = rho_v2_ll / rho_ll
-  v_mag_ll = sqrt(v1_ll^2 + v2_ll^2)
-  p_ll = (equations.gamma - 1) * (rho_e_ll - 1/2 * rho_ll * v_mag_ll^2)
+  # Get the velocity value in the appropriate direction
+  if orientation == 1
+    v_ll = rho_v1_ll / rho_ll
+    v_rr = rho_v1_rr / rho_rr
+  else # orientation == 2
+    v_ll = rho_v2_ll / rho_ll
+    v_rr = rho_v2_rr / rho_rr
+  end
+  # Calculate pressure and sound speed
+  p_ll = pressure(u_ll, equations)
   c_ll = sqrt(equations.gamma * p_ll / rho_ll)
-  v1_rr = rho_v1_rr / rho_rr
-  v2_rr = rho_v2_rr / rho_rr
-  v_mag_rr = sqrt(v1_rr^2 + v2_rr^2)
-  p_rr = (equations.gamma - 1) * (rho_e_rr - 1/2 * rho_rr * v_mag_rr^2)
+  p_rr = pressure(u_rr, equations)
   c_rr = sqrt(equations.gamma * p_rr / rho_rr)
 
-  λ_max = max(v_mag_ll, v_mag_rr) + max(c_ll, c_rr)
+  λ_max = max(abs(v_ll), abs(v_rr)) + max(c_ll, c_rr)
 end
 
 
 @inline function max_abs_speed_naive(u_ll, u_rr, normal_direction::AbstractVector, equations::CompressibleEulerEquations2D)
-  return max_abs_speed_naive(u_ll, u_rr, 0, equations) * norm(normal_direction)
+  rho_ll, rho_v1_ll, rho_v2_ll, _ = u_ll
+  rho_rr, rho_v1_rr, rho_v2_rr, _ = u_rr
+
+  # Calculate normal velocities and sound speed
+  # left
+  v1_ll = rho_v1_ll / rho_ll
+  v2_ll = rho_v2_ll / rho_ll
+  v_ll = (  v1_ll * normal_direction[1]
+          + v2_ll * normal_direction[2] )
+  p_ll = pressure(u_ll, equations)
```

Comment:
This will double some work since some division occur also in `pressure` (if the compiler isn't smart enough). What about
```suggestion
  rho_ll, v1_ll, v2_ll, p_ll = cons2prim(u_ll, equations)
  v_ll = (  v1_ll * normal_direction[1]
          + v2_ll * normal_direction[2] )
```
If that's okay for you, you would also need to remove the lines `rho_ll, rho_v1_ll, rho_v2_ll, _ = u_ll` etc. above and adapt the part for `_rr` below. Basically the same comment applies to the Cartesian version (but you might double check and benchmark yourself).

Reply:
Sure, I can adjust this. I thought I was being so clever using the `pressure` function but I didn't account for the additional divisions. Using `cons2prim` will also make this procedure cleaner